internal/pkg/agent/application/monitoring/v1_monitor.go

```diff
@@ -471,6 +492,13 @@ func (b *BeatsMonitor) injectLogsInput(cfg map[string]interface{}, components []
 				// update event.dataset with the now used data_stream.dataset
 				map[string]interface{}{
 					"copy_fields": map[string]interface{}{
+						"when": map[string]any{
```

Comment:
do we anticipate something to have written in the target event.dataset field? As far as I can tell this condition missing isn't the reason of the reported issue, right?

Reply:
As I mentioned on the other ones, that's just extra caution.

Reply:
Done on 7f3b86f3db